### ISSUE TITLE
[Enhancement] Rule to flag ambiguous YAML syntax

### ISSUE THREAD

**Issue by miggy9001** [adrienverge/yamllint#309]
While rare there are a few syntax issues where the YAML can be valid yet the ambiguity of the syntax will cause issues where parsers can misinterpreted. 

The case I came across is for inline parsing of a string with a comma inside of the inline object:
```Yaml
rootKey:
  - { date: 2020-08-31, stringKey: This is a string, that happens to contain the dreaded comma. }
```

This will pass the YAML lint yet it can be miss interpreted by a YAML parser such as the Ruby YAML parser to pick up the comma as indicating the next string is another object key. 

Here is the YAML spec around this that says the above example is not "invalid" YAML: https://yaml.org/spec/1.2/spec.html#id2788859

To solve this my suggestion is to make a new rule to flag any ambiguous syntax like the example listed above. This could be a simple enable or disable configuration to flag these as warning or error.

**Comment by adrienverge** [comment]
This is interesting!

I think that trying to guess what the user wanted to write goes beyond the limit of what a linter should do (especially since `{key: string, other}` is valid YAML and translates to `{key: string, other: null}`).

However, I see 2 ways to satisfy your need:
1. Use the [`quoted-strings` rule](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.quoted_strings) to force the usage of quotes, hence preventing `,` to be misinterpreted.
   ```yaml
   { stringKey: "This is a string, that happens to contain the dreaded comma." }
   ```
2. Use the [`empty-values` rule](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.empty_values) with `forbid-in-flow-mappings: true`. Indeed,
   ```yaml
   { stringKey: This is a string, that happens to contain the dreaded comma. }
   ```
   translates to
   ```json
   { "stringKey": "This is a string",
     "that happens to contain the dreaded comma.": null }
   ```
   so forbidding the `that happens to contain the dreaded comma.` key to have an implicitly-null value would prevent this notation.

   (The problem is that `empty-values` would need to be fixed: it currently detects `{ key: }`, but not `{ key }`.)

**Comment by miggy9001** [comment]
@adrienverge The quoted string is the temporary fix we have a few trial things with. 

For the second option is there a defect around the issue is for this? We also tried using this but noticed it was not picking the scenario above as an error and was not sure that was a defect or intended. I skimmed through the open defects yesterday and did not see one.

Depending on the scope of the change I would be glad to contribute the changes and test cases back to this project.

**Comment by adrienverge** [comment]
@miggy9001 excuse me but I'm not sure I understood clearly what you mean :confused:
Anyway, in my previous message I meant that:
- the `empty-values` rule should be the one to detect what you want (in my opinion),
- but this rule does not (yet) detect `{ key }` (only `{ key: }`), so it needs to be improved (via a contribution to the project) :+1:

**Comment by miggy9001** [comment]
@adrienverge sorry for the confusion. I was just asking for the detect for `{ key: }` has an issue logged for it already?

**Comment by adrienverge** [comment]
I'm not sure (there are a lot of issues in https://github.com/adrienverge/yamllint/issues?q=is%3Aissue), but I don't think so. But better than an issue, a pull request would be best!
